smslibrary/src/main/java/com/eis/smslibrary/SMSReceivedBroadcastReceiver.java

```diff
@@ -30,30 +33,41 @@ public void onReceive(Context context, Intent intent) {
         if (extras == null) {
             return;
         }
-
-        Object[] smsExtras = (Object[]) extras.get("pdus");
-        String format = (String) extras.get("format");
-
-        if(smsExtras == null) //could be null
+        Object[] pdus = (Object[]) extras.get("pdus");
+        if (pdus == null) {
             return;
+        }
+        String format = (String) extras.get("format");
```

Comment:
no check for android version
not all android version use format

Reply:
It'll just be `null` in older Android versions

Reply:
`format` being `null` is not a problem, because we check the Android version later, before using it

Reply:
@MattiaFanan-uni can I mark this conversation as resolved?